.github/workflows/unit-tests.yml

```diff
@@ -12,9 +12,14 @@ concurrency:
     group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
     cancel-in-progress: true
 
+permissions:
+  contents: read
+
 jobs:
 
   tests:
+    permissions:
+      contents: write  # for Git to git apply
```

Comment:
Why do we need this ? Our unit tests are doing some git-based magic to test complex things, but they don't push those changes to the repository.

Reply:
Hi @stof I misunderstood what `git apply` does. I have removed these permissions.